assets/js/wfparameters/charts/ChartEditor.js

```diff
@@ -37,90 +25,89 @@ function JSONtoCSV(d) {
   }
 }
 
-
 export default class ChartEditor extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       loading: true,
       model: null,      // Chartbuilder model object. Also where we store/update current parameter values
-      inputData: null
+      inputData: null,
     };
+    this.lastWfModuleStatus = null
     this.onChangeChartSettings = this.onChangeChartSettings.bind(this);
     this.onChangeTitle = this.onChangeTitle.bind(this);
-    this.onChangeDate = this.onChangeDate.bind(this);
     this.onChangePrefix = this.onChangePrefix.bind(this);
     this.onChangeSuffix = this.onChangeSuffix.bind(this);
-    this.getStateFromStores = this.getStateFromStores.bind(this);
-    this.saveStateToDatabase = debounce(this.saveStateToDatabase, 1000);
   }
 
-   // Turn ChartBuilder errors into module errors (so user gets the red light etc.)
-  parseErrors(errors) {
-    var errorMessage = errors.messages
-      .filter((m) => {
-        return m.type === 'error';
-      })
-      .map((m) => {
-        return m.text;
-      })
-      .join("\n\r");
-    if (errorMessage !== '') {
+  ickyDispatchSideEffect() {
```

Comment:
Maybe a more descriptive name? It does handle data format errors, after all

Reply:
I'm not sure how it's icky to side effect -- components change the store all the time. 
But name to taste.